packages/alfa-rules/src/common/diagnostic/with-other-heading.ts

```diff
@@ -0,0 +1,138 @@
+import { Diagnostic } from "@siteimprove/alfa-act";
+import { Element } from "@siteimprove/alfa-dom";
+import { Hash } from "@siteimprove/alfa-hash";
+import { Option } from "@siteimprove/alfa-option";
+
+/**
+ * @internal
+ */
+export type HeadingPosition = "previous" | "next" | "unknown";
```

Comment:
**Suggestion:** I think we can stay with only the first two for now, and add more if need arise. Whatever we change later, it's pretty sure that `unknown` won't be the text we want.